Mathlib/Data/Bool/Basic.lean

```diff
@@ -150,8 +135,8 @@ theorem ne_not {a b : Bool} : a ≠ !b ↔ a = b :=
   not_eq_not
 #align bool.ne_bnot Bool.ne_not
 
-theorem not_ne : ∀ {a b : Bool}, (!a) ≠ b ↔ a = b := not_not_eq
-#align bool.bnot_ne Bool.not_ne
+@[deprecated (since := "2024-06-07")] alias not_ne := not_not_eq
```

Comment:
Note that these lemmas aren't syntactically equal: one uses `Not (_ = _)`, another uses `Ne _ _`. Not sure if it justifies leaving the `ne` lemma.

Reply:
Given that it isn't used, I'd prefer to err on the side of more aggressively cleaning up. If someone downstream is upset by the deprecation they still have a chance to say something about it.